test/functional/wallet_listsinceblock.py

```diff
@@ -108,23 +111,25 @@ def test_reorg(self):
         senttx = self.nodes[2].sendtoaddress(self.nodes[0].getnewaddress(), 1)
 
         # generate on both sides
-        lastblockhash = self.nodes[1].generate(6)[5]
-        self.nodes[2].generate(7)
-        self.log.info('lastblockhash=%s' % (lastblockhash))
+        lastblockhash_nodes_1 = self.nodes[1].generate(6)[5]
+        firstblockhash_nodes_2 = self.nodes[2].generate(7)[0]
+        self.log.debug("lastblockhash nodes[1]={}".format(lastblockhash_nodes_1))
+        self.log.debug("firstblockhash nodes[2]={}".format(firstblockhash_nodes_2))
 
         self.sync_all(self.nodes[:2])
         self.sync_all(self.nodes[2:])
 
         self.join_network()
 
-        # listsinceblock(lastblockhash) should now include tx, as seen from nodes[0]
-        lsbres = self.nodes[0].listsinceblock(lastblockhash)
-        found = False
-        for tx in lsbres['transactions']:
+        # listsinceblock(lastblockhash_nodes_1) should now include tx as seen from nodes[0],
+        # and return the block height which listsinceblock now exposes since a5e7795.
+        transactions = self.nodes[0].listsinceblock(lastblockhash_nodes_1)['transactions']
+        for tx in transactions:
             if tx['txid'] == senttx:
-                found = True
+                found = tx
                 break
-        assert found
+        assert_equal(found['blockheight'],
```

Comment:
This will throw `NameError` if no match is found above? :)

Reply:
Open to suggestions on making it fail more gracefully, but IIUC it shouldn't fail?

Reply:
Could add `found = {'blockheight': -1}` if you would prefer it to fail with `AssertionError: not(0 == 103)` instead of `NameError: name 'found' is not defined`?

Reply:
Perhaps `found = None` and then `assert found is not None` on the line before the `assert_equal`? :)

Reply:
Done, LMK if you prefer a different initial value :)

Reply:
You've added 2 lines! :smile: ... SGTM, done

Reply:
nit: You can also use `next()`, as we do in other tests.

Reply:
> nit: You can also use `next()`, as we do in other tests.

agree, will look at it

Reply:
@MarcoFalke something like this?
```diff
-        found = None
-        for tx in transactions:
-            if tx['txid'] == senttx:
-                found = tx
-                break
-        assert found is not None
+        found = next(tx for tx in transactions if tx['txid'] == senttx)
````